### PR TITLE
audit: 4.0 -> 4.0.2

### PR DESCRIPTION
**Pull request by itepastra**
Updated audit.
had to add `lib` to the output directories to fix the loop in the build results.

The patches got merged upstream so I removed those

I also changed the deprecated `--replace` to `--replace-fail` and used nixfmt to format it.

close #350628

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by samueltardieu** [comment]
> had to remove `bin` from the output directories to fix the build, all binaries are still in the result.

I think it would be better to keep `bin` first and add `lib` to host the dynamic libraries. That would also have broken the loop, and no package in the tree depends on `audit.out` so the dynamic libraries being moved should have had no consequences.

Here, you are making `dev` the default output, and you put the binaries in there, none of which is usual.

Also, here is a link to the update request: https://github.com/NixOS/nixpkgs/issues/350628

**Review by samueltardieu — APPROVED** [review]

**Comment by samueltardieu** [comment]
Could you put a "Close" or "Fix" for the update request so that it get closed when this package is merged?

**Review by emilazy — APPROVED** [review]

File: pkgs/by-name/au/audit/package.nix [hunk 1]
```diff
@@ -1,64 +1,58 @@
...
-      url = "https://github.com/linux-audit/audit-userspace/commit/4192eb960388458c85d76e5e385cfeef48f02c79.patch";
-      hash = "sha256-G6CJ9nBJSsTyJ0qq14PVo+YdInAvLLQtXcR25Q8V5/4=";
-    })
-  ];
-
   postPatch = ''
     substituteInPlace bindings/swig/src/auditswig.i \
-      --replace "/usr/include/linux/audit.h" \
+      --replace-fail "/usr/include/linux/audit.h" \
                 "${linuxHeaders}/include/linux/audit.h"
```
Comment (emilazy) [inline pkgs/by-name/au/audit/package.nix:31]: Would prefer to see the alignment here fixed or removed. But not a blocker.

**Review by emilazy — APPROVED** [review]

File: pkgs/by-name/au/audit/package.nix (hunk 1 above)
Comment (itepastra) [reply pkgs/by-name/au/audit/package.nix:31]: Will do tomorrow

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Pull request merged**